docs/standard/serverless-architecture/architecture-approaches/architecture-deployment-approaches.md

```diff
@@ -0,0 +1,167 @@
+* Direct deployment and improved DevOps, CI, and CD pipelines
+* Automatic upgrades, updates, and security patches
+* Push-button scale out and scale up (elastic scale)
+
+The main disadvantage of PaaS traditionally has been vendor lock-in. For example, some PaaS providers only support ASP.NET, Node.js, or other specific languages and platforms. Products like Azure App Service have evolved to address multiple platforms and support a variety of languages and frameworks for hosting web apps.
+
+![Platform-as-a-Service Architecture](./media/architecture-deployment-approaches/paas-architecture.png)
+
+## Software-as-a-Service (SaaS)
+
+Software-as-a-Service or SaaS is centrally hosted and available without local installation or provisioning. SaaS often is hosted on top of PaaS as a platform for deploying software. SaaS provides services to run and connect with existing software. SaaS is often industry and vertical specific. SaaS is often licensed and typically provides a client/server model. Most modern SaaS offerings use web-based apps for the client. Companies typically consider SaaS as a business solution to license offerings. It is not often implemented as architecture consideration for scalability and maintainability of an application. Indeed, most SaaS solutions are built on IaaS, PaaS, and/or serverless back-ends.
+
+Learn more about SaaS through a [sample application](/azure/sql-database/saas-tenancy-welcome-wingtip-tickets-app).
+
+## Containers and Functions-as-a-Service (FaaS)
+
+Containers are an interesting solution that enables PaaS-like benefits without the IaaS overhead. A container is essentially a runtime that contains the bare essentials needed to run a unique application. The kernel or core part of the host operating system and services such as storage are shared across a host. The shared kernel enables containers to be lightweight (some are mere megabytes in size, compared to the gigabyte size of typical virtual machines). With hosts already running, containers can be started quickly, facilitating high availability. The ability to spin up containers quickly also provides extra layers of resiliency. Docker is one of the more popular implementations of containers.
+
+Benefits of containers include:
+
+* Lightweight and portable
+* Self-contained so no need to install dependencies
+* Provide a consistent environment regardless of the host (runs exactly same on a laptop as on a cloud server)
+* Can be provisioned quickly for scale-out
+* Can be restarted quickly to recover from failure
+
+A container runs on a container host (that in turn may run on a bare metal machine or a virtual machine). Multiple containers or instances of the same containers may run on a single host. For true failover and resiliency, containers must be scaled across hosts.
+
+For more information about Docker containers, read: [What is Docker?](/dotnet/standard/microservices-architecture/container-docker-introduction/docker-defined)
```

Comment:
should use relative paths for in-repo links for the build to validate if the link exists or not
`..\..\microservices-architecture/container-docker-introduction/docker-defined.md`